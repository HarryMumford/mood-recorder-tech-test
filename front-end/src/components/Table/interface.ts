export interface TableRowProps {
  readonly data: string;
}
