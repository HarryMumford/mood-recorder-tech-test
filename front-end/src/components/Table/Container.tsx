import * as React from 'react';
import styled from 'styled-components';

const Table = () => {
  const TableContainer = styled.div`
    width: 200px;
    background-color: lightpink;
  `;
  return <TableContainer>Table goes here</TableContainer>;
};

export default Table;
