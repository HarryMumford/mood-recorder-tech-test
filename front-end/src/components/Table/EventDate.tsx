import * as React from 'react';
import { getDay, getDate, getMonth } from '../../services/dateFormatter';
import styled from 'styled-components';

const FullDate = styled.div`
  font-size: 0.7rem;
  color: darkslategray;
`;

const EventDate = (props: { timestamp: string }) => {
  const { timestamp } = props;
  const day = getDay(timestamp);
  const date = getDate(timestamp);
  const month = getMonth(timestamp);
  const year = new Date(timestamp).getFullYear();

  return (
    <>
      <FullDate key={timestamp}>{`${day} ${date} ${month} ${year}`}</FullDate>
    </>
  );
};

export default EventDate;
