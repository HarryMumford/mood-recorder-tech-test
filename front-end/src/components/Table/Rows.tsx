import * as React from 'react';
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { RootStore } from '../../flux/store';
import { getConcernEvents } from '../../flux/actions/concernEvents';
import EventDate from './EventDate';
import Subtitle from '../Chart/Subtitle';
import { TableRowProps } from './interface';
import { severityColors } from '../../assets/constants/colors';
import styled from 'styled-components';

const SeverityWord = styled.td<TableRowProps>`
  color: ${(props) => props.data};
  font-size: 1rem;
  font-weight: bold;
`;

const TableHeader = styled.tr`
  font-weight: bold;
  color: darkslategray;
  font-size: 0.7rem;
`;

const TableData = styled.td`
  padding: 10px;
  text-align: left;
`;

const Rows = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getConcernEvents());
  }, []);

  const concernEventState = useSelector(
    (state: RootStore) => state.concernEvent
  );

  const concernEvents = concernEventState.concernEvent;

  return (
    <>
      <Subtitle>Latest Incidents</Subtitle>
      <table>
        <TableHeader>
          <TableData>Severity</TableData>
          <TableData>Observer</TableData>
          <TableData>Date</TableData>
        </TableHeader>
        <tbody>
          {concernEvents &&
            concernEvents.map((ce) => {
              const { timestamp, severity, observer } = ce;
              const color = severityColors[severity];

              return (
                <tr key={timestamp}>
                  <SeverityWord data={color}>{severity}</SeverityWord>
                  <TableData>{observer}</TableData>
                  <EventDate timestamp={timestamp} />
                </tr>
              );
            })}
        </tbody>
      </table>
    </>
  );
};

export default Rows;
