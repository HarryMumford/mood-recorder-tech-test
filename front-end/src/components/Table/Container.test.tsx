import * as React from 'react';
import { shallow } from 'enzyme';
import Table from './Container';

describe('Table container', () => {
  it('should render the container correctly', () => {
    const component = shallow(<Table />);

    expect(component).toMatchSnapshot();
  });
});
