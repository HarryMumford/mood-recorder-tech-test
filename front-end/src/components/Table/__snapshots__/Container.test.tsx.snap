// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<styled.div>
  <Rows />
</styled.div>
`;

exports[`Table container should render the container correctly 1`] = `
<styled.div>
  <Rows />
</styled.div>
`;

exports[`Table should render correctly 1`] = `
<styled.div>
  <Rows />
</styled.div>
`;
